app/src/main/java/org/schabi/newpipe/MainActivity.java

```diff
@@ -633,23 +633,20 @@ private void updateDrawerNavigation() {
 
         final Fragment fragment = getSupportFragmentManager().findFragmentById(R.id.fragment_holder);
 
-        //If it's MainFragment enable the Navigation Drawer again
-        if (fragment instanceof MainFragment) {
-            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
-            if (toggle != null) {
-                toggle.syncState();
-                toolbar.setNavigationOnClickListener(v -> drawer.openDrawer(GravityCompat.START));
-                drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNDEFINED);
-            }
-            return;
-        }
-
         //We need to enable the home button for the search fragment to go back to MainFragment
         if (fragment instanceof SearchFragment) {
             if (toggle != null) {
                 drawer.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
                 getSupportActionBar().setDisplayHomeAsUpEnabled(true);
                 toolbar.setNavigationOnClickListener(v -> onHomeButtonPressed());
+                return;
+            }
+        }else{
+            getSupportActionBar().setDisplayHomeAsUpEnabled(false);
```

Comment:
Why is this outside of the if here, but inside in the part above?

Reply:
Which part @Stypox you meant before the bug araised?

Reply:
Before the bug, I assumed that to go to VideoDetailsFragment the MainFragment is the only way, but when search fragment is called, the video details fragment is gone. This time, I activated the back button only to search, enabled the menu to the rest of the fragments. But settings and About will not be affected.